apps/tour-of-heroes-ngxs/src/app/hero-detail/hero-detail.component.ts

```diff
@@ -27,21 +27,19 @@ export class HeroDetailComponent {
 
   constructor(
     private route: ActivatedRoute,
-    private heroState: HeroStateService,
+    private heroState: HeroStateFacade,
     private location: Location,
     private state: RxState<HeroDetailComponentState>,
     public configService: ConfigService
   ) {
     const hero$ = this.route.paramMap.pipe(
-      switchMap((params) =>
-        this.heroState.hero$(parseInt(params.get('id'), 10))
-      )
+      switchMap(params => this.heroState.hero$(parseInt(params.get('id'), 10)))
     );
     this.state.connect('hero', hero$);
     const saveEffect$ = this.save$.pipe(
       withLatestFrom(this.change$),
       map(([oldHero, change]) => ({ ...oldHero, ...change })),
-      switchMap((hero) => this.heroState.dispatchUpdateHero(hero)),
+      switchMap(hero => this.heroState.dispatchUpdateHero(hero)),
```

Comment:
You could also do something like this:

this.state.hold(saveTrigger$, hero => {
  this.heroState.dispatchUpdateHero(hero);
  this.goBack();
});